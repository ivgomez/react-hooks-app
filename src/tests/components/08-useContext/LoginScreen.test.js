import "@testing-library/jest-dom";
import React from "react";
import { mount } from "enzyme";
import { LoginScreen } from "../../../components/08-useContext/LoginScreen";
import { UserContext } from "../../../components/08-useContext/UserContext";

describe("Test on <LoginScreen/>", () => {
  const setUser = jest.fn();
  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );
  test("should rendered right ", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should run the setUser with the argument ", () => {
    wrapper.find("button").prop("onClick")();
    expect(setUser).toHaveBeenCalledWith({ id: 123, name: "papaleta" });
  });
});
