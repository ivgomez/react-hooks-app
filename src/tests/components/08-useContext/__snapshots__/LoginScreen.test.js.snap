// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <LoginScreen/> should rendered right  1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
